api/v1beta1/types.go

```diff
@@ -579,6 +579,37 @@ type SubnetSpec struct {
 	SubnetClassSpec `json:",inline"`
 }
 
+// Network Interfaces to attach to each VM
```

Comment:
```suggestion
// AzureNetworkInterface defineds a network interface to attach to each VM.
```
Typically we want to start the documentation comment with the name of the type/function. I believe the linters will complain about it if it doesn't match.